packages/cln_plugin/lib/src/cln_plugin_base.dart

```diff
@@ -162,11 +169,22 @@ class Plugin implements CLNPlugin {
     configurePlugin();
   }
 
-  Future<Map<String, Object>> _call(
+  Future<Map<String, Object>> _callHook(
+      String name, Map<String, Object> request) async {
+    if (hooks.containsKey(name)) {
+      var method = hooks[name]!;
+      return await method.call(this, request);
+    }
+    throw Exception("Hook with name $name not found!");
+  }
+
+  Future<Map<String, Object>> _callRPCMethod(
       String name, Map<String, Object> request) async {
     if (rpcMethods.containsKey(name)) {
       var method = rpcMethods[name]!;
       return await method.call(this, request);
+    } else if (hooks.containsKey(name)) {
+      return await _callHook(name, request);
```

Comment:
BTW thinking that in this way we are hinding the error in case of the wrong hook, but we still do not know if it is a hooks call.

Soo lets give a try